### PR TITLE
Merge Sort in Kotlin

### LINKED ISSUE #2136: Merge Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart
- [x] Go
- [x] Java
- [x] JS
- [ ] Kotlin - @shubhammantri1
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [x] Rust - @m-code12
- [x] TypeScript - @m-code12

### PR DESCRIPTION
**Pull request by shubhammantri1**
Fixes #2136 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Merge Sort in Kotlin

#### Languages Used:
- Merge Sort in Kotlin

#### Files Added:
- Merge_Sort.kt

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by shubhammantri1** [comment]
@jainaman224 Please review it.

File: Merge_Sort/Merge_Sort.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            index1++
+        }
+        else
+        {
+            arr[k] = arr2[index2]
+            index2++
+        }
+        k++
+    }
+    //Inserting remaining element
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.kt]: Add space after `//`.

File: Merge_Sort/Merge_Sort.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        arr[i] = read.nextLine().toInt()
+    }
+    mergeSort(arr, 0, arrSize-1)
+    println("Sorted Array is:")
+    for (i in 0 until arrSize)
+    {
+        print("${arr[i]} ")
+    }
+}
+/*
```
Comment (jainaman224) [inline Merge_Sort/Merge_Sort.kt]: Add new line above this.

**Comment by shubhammantri1** [comment]
@jainaman224 Please Review again

File: Merge_Sort/Merge_Sort.kt (hunk 2 above)
Comment (shubhammantri1) [reply Merge_Sort/Merge_Sort.kt]: Done.

File: Merge_Sort/Merge_Sort.kt (hunk 1 above)
Comment (shubhammantri1) [reply Merge_Sort/Merge_Sort.kt]: Done. @jainaman224

**Comment by shubhammantri1** [comment]
It's been more than 2 days please review my PR. @jainaman224 @SakshiShreya

**Comment by raghav-dalmia** [comment]
Looks good to me. Squash your commits. I will merge it after that.

**Comment by somya-kapoor** [comment]
@shubhammantri1  Are you working on it?

**Comment by shubhammantri1** [comment]
Yes, I Completed my work. @somya-kapoor

**Comment by somya-kapoor** [comment]
Please squash your commits

**Comment by shubhammantri1** [comment]
@somya-kapoor @raghav-dalmia Please review it. I rebase my commits.

**Pull request closed without merging**